File: 13/two.lua
-- --- Advent of code 2021: Day 13 ---

-- (Skeleton file automatically created by aocTool, developed by Bolu, 2020-2021.)

package.path = package.path .. ";../modules/?.lua"
local tools = require "tools"

input = tools.readlines("input.txt")

dots = {}
folds = {}
reading_folds = false
for _, line in pairs(input)do
	if reading_folds then
		fold = line:sub(#"fold along _")
		folds[#folds + 1] = fold
	else
		if #line == 0 then
			reading_folds = true
		else
			dots[line] = true
		end
	end
end

for _, fold in pairs(folds) do
	axis, coord = table.unpack(tools.split(fold, "="))
	coord = tonumber(coord)

	new_dots = {}
	for dot, _ in pairs(dots) do
		x, y = table.unpack(tools.map(tonumber, tools.split(dot, ",")))

		if axis == "x" then
			if x < coord then
				new_dots[tostring(x) .. "," .. tostring(y)] = true
			elseif x > coord then
				new_dots[tostring((coord - (x % coord)) % coord) .. "," .. tostring(y)] = true
			end
		else
			if y < coord then
				new_dots[tostring(x) .. "," .. tostring(y)] = true
			elseif y > coord then
				new_dots[tostring(x) .. "," .. tostring((coord - (y % coord)) % coord)] = true
			end
		end
	end
	dots = new_dots
end

x = {}
y = {}

for dot, _ in pairs(dots) do
	_x, _y = table.unpack(tools.map(tonumber, tools.split(dot, ",")))
	x[#x + 1] = _x
	y[#y + 1] = _y
end

str = ""
for j = 0, math.max(table.unpack(y)) do
	for i = 0, math.max(table.unpack(x)) do
		if dots[tostring(i) .. "," .. tostring(j)] then
			str = str .. "#"
		else
			str = str .. " "
		end
	end
	str = str .. "\n"
end

print(str)

File: 13/two.py
# --- Advent of code 2021: Day 13 ---

# (Skeleton file automatically created by aocTool, developed by Bolu, 2020-2021.)

dots, folds = [e.splitlines() for e in open("input.txt").read().split("\n\n")]

dots = set(map(lambda x: (int(x.split(",")[0]), int(x.split(",")[1])), dots))
folds = [(x[len("fold along ")], int(x[len("fold along x="):])) for x in folds]

for axis, coord in folds:
	new_dots = set()
	for x, y in dots:
		if axis == 'x':
			if x < coord:
				new_dots.add((x, y))
			elif x > coord:
				new_dots.add((((coord - (x % coord)) % coord), y))
		else:
			if y < coord:
				new_dots.add((x, y))
			elif y > coord:
				new_dots.add((x, ((coord - (y % coord)) % coord)))
	dots = new_dots

x = [e[0] for e in dots]
y = [e[1] for e in dots]

for j in range(max(y) + 1):
	for i in range(max(x) + 1):
		if (i, j) in dots:
			print("#", end="")
		else:
			print(" ", end="")
	print()
